definitions/checks/check_hotfix_installed.rb

```diff
@@ -0,0 +1,84 @@
+class Checks::CheckHotfixInstalled < ForemanMaintain::Check
+  metadata do
+    label :check_hotfix_installed
+    description 'Check to verify if any hotfix installed on system'
+    preparation_steps do
+      Procedures::Packages::Install.new(:packages => %w[yum-utils])
+    end
+
+    confine do
+      feature(:downstream)
+    end
+  end
+
+  def run
+    if feature(:downstream) && feature(:downstream).subscribed_using_activationkey?
+      skip 'Your system is subscribed using custom activationkey'
+    else
+      with_spinner('Checking for presence of hotfix(es). It may take some time to verify.') do
+        hotfix_rpmlist, installed_pkg_list = installed_packages
+        files_modifications = installed_pkg_list.flat_map { |pkg| modified_files(pkg) }
+        assert(hotfix_rpmlist.empty? && files_modifications.empty?,
+               warning_message(hotfix_rpmlist, files_modifications))
+      end
+    end
+  end
+
+  private
+
+  def modified_files(package)
+    changed_files = []
+    IO.popen(['rpm', '-V', package]) do |pipe|
+      pipe.each do |line|
+        arr_output = line.chomp.split
+        flags = arr_output.first
+        filename = arr_output.last
+        changed_files << filename if flags.include?('5') && filename =~ /\.(rb|py|erb|js)$/
+      end
+    end
+    changed_files
+  end
+
+  def installed_packages
```

Comment:
I got another way to get these details,

1. To get packages from satellite and rhscl repos you can avoid if statement for each package name,

> yumdb search from_repo *satellite* *rhscl*
rh-ruby23-runtime-2.2-7.el7.x86_64
     from_repo = rhel-server-rhscl-7-rpms
rubygem-ansi-1.4.3-3.el7sat.noarch
     from_repo = rhel-7-server-satellite-6.3-rpms

The **yumdb** comes with **yum-utils** only, and this is very quick :)

> time yumdb search from_repo *satellite* *rhscl*
real	0m0.563s
user	0m0.356s
sys	0m0.197s

We just need to disable yum plugins before query and enable it after getting the packages to make it faster else its unnecessary waits for plugins which we dont use while querying yum database,

to disable,

> sed -i -e 's/plugins=1/plugins=0/g' /etc/yum.conf

to enable,

> sed -i -e 's/plugins=0/plugins=1/g' /etc/yum.conf

2. To get the hotfix packages installed you can use,

> rpm -qa release="*HOTFIX*"

Reply:
I have updated command to get hotfix rpms using `rpm -qa release="*HOTFIX*"` command in pull-request.
Kept `repoquery` command as `yumdb` is not giving packages names but having extra input lines.